### ISSUE TITLE
Error loading "{dll}" or one of its dependencies after following windows installation instructions

### ISSUE THREAD

**Issue by zippererer** [intel/intel-extension-for-pytorch#632]
### Describe the bug

I followed the instructions from here:
https://intel.github.io/intel-extension-for-pytorch/#installation?platform=gpu&version=v2.1.30%2bxpu&os=windows&package=pip

Executing 'import intel_extension_for_pytorch as ipex' returns:
C:\ProgramData\anaconda3\Lib\site-packages\torchvision\io\image.py:13: UserWarning: Failed to load image Python extension: 'Could not find module 'C:\ProgramData\anaconda3\Lib\site-packages\torchvision\image.pyd' (or one of its dependencies). Try using the full path with constructor syntax.'If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\ProgramData\anaconda3\Lib\site-packages\intel_extension_for_pytorch\__init__.py", line 91, in <module>
    raise err
OSError: [WinError 126] The specified module could not be found. Error loading "C:\ProgramData\anaconda3\Lib\site-packages\intel_extension_for_pytorch\bin\intel-ext-pt-gpu.dll" or one of its dependencies.

I used procmon to determine why this fails, which was due to 2 errors:

1) C:\ProgramData\anaconda3\Library\bin\ is searched for the mkl_*.dll, and all of those dlls are version 2023.1.1.0 in my conda environment. Removing those was sufficient to find the proper dlls, but completely destroying and recreating the environment would be the recommend fix. I'm guessing upgrading from 2023 -> 2024 did not properly handle these.

2) PATH does not contain oneAPI\compiler\, which contains some of the dependent dlls for the extension.  Adding set PATH=C:\Program Files (x86)\Intel\oneAPI\compiler\2024.1\bin;%PATH% allows the code that iterates PATH in __init__.py to properly find the dlls needed. Alternatively, 'call "%MKLROOT%\..\..\compiler\latest\env\vars.bat"' can be used.

### Versions

broken environment:
C:\ProgramData\anaconda3\Lib\site-packages\torchvision\io\image.py:13: UserWarning: Failed to load image Python extension: 'Could not find module 'C:\ProgramData\anaconda3\Lib\site-packages\torchvision\image.pyd' (or one of its dependencies). Try using the full path with constructor syntax.'If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
Collecting environment information...
PyTorch version: 2.1.0.post2+cxx11.abi
PyTorch CXX11 ABI: No
IPEX version: N/A
IPEX commit: N/A
Build type: N/A

OS: Microsoft Windows 11 Pro
GCC version: N/A
Clang version: N/A
IGC version: N/A
CMake version: version 3.28.1
Libc version: N/A

Python version: 3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:26:23) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22631-SP0
Is XPU available: N/A
DPCPP runtime version: N/A
MKL version: N/A
GPU models and configuration:
N/A
Intel OpenCL ICD version: N/A
Level Zero version: N/A

CPU:
Architecture=9
CurrentClockSpeed=2300
DeviceID=CPU0
Family=198
L2CacheSize=11776
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=2300
Name=12th Gen Intel(R) Core(TM) i7-12700H
ProcessorType=3
Revision=

Versions of relevant libraries:
[pip3] intel-extension-for-pytorch==2.1.30+xpu
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.4
[pip3] numpydoc==1.5.0
[pip3] torch==2.1.0.post2+cxx11.abi
[pip3] torchaudio==2.1.0.post2+cxx11.abi
[pip3] torchvision==0.16.0.post2+cxx11.abi
[conda] _anaconda_depends         2023.09             py311_mkl_1
[conda] blas                      1.0                         mkl
[conda] intel-extension-for-pytorch 2.1.30+xpu               pypi_0    pypi
[conda] mkl                       2023.1.0         h6b88ed4_46358
[conda] mkl-service               2.4.0           py311h2bbff1b_1
[conda] mkl_fft                   1.3.8           py311h2bbff1b_0
[conda] mkl_random                1.2.4           py311h59b6b97_0
[conda] numpy                     1.24.4                   pypi_0    pypi
[conda] numpydoc                  1.5.0           py311haa95532_0
[conda] torch                     2.1.0.post2+cxx11.abi          pypi_0    pypi
[conda] torchaudio                2.1.0.post2+cxx11.abi          pypi_0    pypi
[conda] torchvision               0.16.0.post2+cxx11.abi          pypi_0    pypi

with fixes applied:
C:\ProgramData\anaconda3\Lib\site-packages\torchvision\io\image.py:13: UserWarning: Failed to load image Python extension: 'Could not find module 'C:\ProgramData\anaconda3\Lib\site-packages\torchvision\image.pyd' (or one of its dependencies). Try using the full path with constructor syntax.'If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
Collecting environment information...
PyTorch version: 2.1.0.post2+cxx11.abi
PyTorch CXX11 ABI: No
IPEX version: 2.1.30+xpu
IPEX commit: 474a6b3cb
Build type: Release

OS: Microsoft Windows 11 Pro
GCC version: N/A
Clang version: 17.0.3
IGC version: 2024.1.0 (2024.1.0.20240308)
CMake version: version 3.28.1
Libc version: N/A

Python version: 3.11.5 | packaged by Anaconda, Inc. | (main, Sep 11 2023, 13:26:23) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.22631-SP0
Is XPU available: True
DPCPP runtime version: N/A
MKL version: N/A
GPU models and configuration:
[0] _DeviceProperties(name='Intel(R) Iris(R) Xe Graphics', platform_name='Intel(R) Level-Zero', dev_type='gpu', driver_version='1.3.29283', has_fp64=0, total_memory=30151MB, max_compute_units=96, gpu_eu_count=96)
[1] _DeviceProperties(name='Intel(R) Arc(TM) A770M Graphics', platform_name='Intel(R) Level-Zero', dev_type='gpu', driver_version='1.3.29283', has_fp64=0, total_memory=15930MB, max_compute_units=512, gpu_eu_count=512)
Intel OpenCL ICD version: N/A
Level Zero version: N/A

CPU:
Architecture=9
CurrentClockSpeed=2300
DeviceID=CPU0
Family=198
L2CacheSize=11776
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=2300
Name=12th Gen Intel(R) Core(TM) i7-12700H
ProcessorType=3
Revision=

Versions of relevant libraries:
[pip3] intel-extension-for-pytorch==2.1.30+xpu
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.4
[pip3] numpydoc==1.5.0
[pip3] torch==2.1.0.post2+cxx11.abi
[pip3] torchaudio==2.1.0.post2+cxx11.abi
[pip3] torchvision==0.16.0.post2+cxx11.abi
[conda] _anaconda_depends         2023.09             py311_mkl_1
[conda] blas                      1.0                         mkl
[conda] intel-extension-for-pytorch 2.1.30+xpu               pypi_0    pypi
[conda] mkl                       2023.1.0         h6b88ed4_46358
[conda] mkl-service               2.4.0           py311h2bbff1b_1
[conda] mkl_fft                   1.3.8           py311h2bbff1b_0
[conda] mkl_random                1.2.4           py311h59b6b97_0
[conda] numpy                     1.24.4                   pypi_0    pypi
[conda] numpydoc                  1.5.0           py311haa95532_0
[conda] torch                     2.1.0.post2+cxx11.abi          pypi_0    pypi
[conda] torchaudio                2.1.0.post2+cxx11.abi          pypi_0    pypi
[conda] torchvision               0.16.0.post2+cxx11.abi          pypi_0    pypi

**Comment by wangkl2** [comment]
@zippererer Thanks for reporting this. From your description, with the fix you mentioned, it works fine now with importing ipex, right?
IPEX 2.1.30+xpu works with oneAPI 2024.1 (with both oneMKL and DPCPP as the dependency). And exactly if you've installed IPEX before, we recommend creating a new conda env to avoid any lib conflict issues. 
At the end of the installation guide of https://intel.github.io/intel-extension-for-pytorch/#installation?platform=gpu&version=v2.1.30%2bxpu&os=windows&package=pip, it tells to activate both DPCPP and MKL env with the `vars.bat`. From your output in the broken env, either DPCPP version or MKL version is found. And that's why ipex module cannot be loaded and XPU cannot be fetched.

**Comment by zippererer** [comment]
@wangkl2 yes, I am able to import.
I ran both the vars.bat for DPCPPROOT and MKLROOT for both the broken and fixed environments. For the fixed environment, I had to also run the compiler vars.bat (or manually add to the path), otherwise the import for the extension would fail.

I see the issue (and confusion) now, and its on my end. The DPCPPROOT is set to oneAPI\dpcpp-ct\latest, and documentation says to point it at the compiler\latest. I've updated those on my machine.

Thanks for your help!
